src/features/cluster/queries/getClusterInfoQuery.ts

```diff
@@ -3,18 +3,17 @@ import { queryKeys } from '@/react-query/constants';
 import { queryOptions } from '@tanstack/react-query';
 import { Cluster } from '@/lib/api.patch';
 
-async function getClusterInfo(context: { queryKey: (string | undefined)[] }) {
-	const { data } = await apiClient.get(`/Cluster/${context.queryKey[1]}` as '/Cluster/{id}');
+export async function getClusterInfo(clusterId: string) {
+	const { data } = await apiClient.get(`/Cluster/${clusterId}` as '/Cluster/{id}');
 	return data as unknown as Cluster;
 }
 
-function getClusterInfoQueryOptions(clusterId?: string) {
+export function getClusterInfoQueryOptions(clusterId?: string, refetch?: boolean) {
 	return queryOptions({
 		queryKey: [queryKeys.cluster, clusterId],
-		queryFn: getClusterInfo,
+		queryFn: () => getClusterInfo(clusterId!),
```

Comment:
I keep going back and forth on this pattern... on the one hand, I really like not creating more contexts. By using the passed `context.queryKey`, memory management becomes simpler for the runtime. But as soon as we go to utilize `getClusterInfo` from somewhere else in the codebase, it's much harder to call...

Reply:
Hmmmmm 🤔, while memory management becomes simpler for the runtime, does the overhead of creating more context outweigh the simplicity of being able to utilize this elsewhere in the codebase? I personally say no but it's definitely something for us to be aware of as we improve the standardize patterns in the codebase overtime